### PR TITLE
Feature/wma 134/ user roles

### PR DESCRIPTION
**Pull request by Caleb-Nieto**
### Feature / Bug Description
Feature/wma 134/ user roles

### Solution
Notfound error page created, this page will display when you type an inexist url
Unauthorized page created, this page will display when you type an incorrect url or you try to acces to an url with other role
In the Login was implemented other roles to redirect to the corresponding main page of each role
For sysAdmin role were implemented buttons in the sider options since its access is from its site

### Notes
no notes needed

### Tickets
[WMA-134](https://cdentalcaregroup.atlassian.net/browse/WMA-134)

### Screenshots / Screencasts
Captura de pantalla 2024-08-11 180954
Captura de pantalla 2024-08-11 180941
Captura de pantalla 2024-08-11 180928
Captura de pantalla 2024-08-11 180920
Captura de pantalla 2024-08-11 180910

[WMA-134]: https://cdentalcaregroup.atlassian.net/browse/WMA-134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/charts/Charts.tsx [hunk 1]
```diff
@@ -108,7 +113,7 @@ const Charts = () => {
                   className="md:flex-1 w-full mx-auto bg-gray-100 rounded-xl shadow-md"
                 >
                   <div className="w-full h-60">
-                    <AreasChart siteId={siteId} />
+                    <AreasChart siteId={location?.state?.siteId} />
```
Comment (imadiaz) [inline src/pages/charts/Charts.tsx:116]: what do you think if you move this into a variable?
something like

  const sideId = location.state.sidteId?

File: src/pages/errors/NotFound.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Button, Result } from "antd";
+import { Link } from "react-router-dom";
+
+const NotFound: React.FC = () => (
+  <Result
+    status="404"
+    title="404"
+    subTitle="Sorry, the page you visited does not exist."
```
Comment (imadiaz) [inline src/pages/errors/NotFound.tsx:9]: can we move this into a strings?

File: src/pages/errors/NotFound.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    extra={
+      <Link to={"/"}>
+        <Button type="primary">Go back</Button>
```
Comment (imadiaz) [inline src/pages/errors/NotFound.tsx:12]: same here

File: src/pages/errors/Unauthorized.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import { Button, Result } from "antd";
+import { Link } from "react-router-dom";
+
+const Unauthorized: React.FC = () => (
+  <Result
+    status="403"
+    title="403"
+    subTitle="Sorry, you are not authorized to access this page."
```
Comment (imadiaz) [inline src/pages/errors/Unauthorized.tsx:9]: same here

File: src/pages/errors/Unauthorized.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    extra={
+      <Link to={"/"}>
+        <Button type="primary">Go back</Button>
```
Comment (imadiaz) [inline src/pages/errors/Unauthorized.tsx:12]: and here

File: src/pages/priority/Priorities.tsx [hunk 6]
```diff
@@ -125,7 +120,10 @@ const Priorities = () => {
     <>
       <div className="h-full flex flex-col">
         <div className="flex flex-col gap-2 items-center m-3">
-          <PageTitle mainText={Strings.prioritiesOf} subText={siteName} />
+          <PageTitle
+            mainText={Strings.prioritiesOf}
+            subText={location?.state?.siteName}
```
Comment (imadiaz) [inline src/pages/priority/Priorities.tsx:125]: same variable for this value?

File: src/pages/routes/Routes.tsx [hunk 7]
```diff
@@ -3,129 +3,288 @@ import { MenuProps } from "antd";
...
+const sysAdminCardTypes = new Route(
+  "Card types",
+  "cardtypes",
+  Routes.SysadminPrefix + Routes.CardTypesAllBySite,
+  <CardTypess rol={UserRoles.SYSADMIN} />,
+  <BiCategory />
+);
+
+const sysAdminCards = new Route(
+  "Cards",
```
Comment (imadiaz) [inline src/pages/routes/Routes.tsx:182]: should we use constants for this?

Comment (Caleb-Nieto) [reply src/pages/routes/Routes.tsx:182]: What do you mean? do you mean about the "Cards" string?

**Review by imadiaz — APPROVED** [review]

**Pull request merged**